alicloud/resource_alicloud_ots_table.go

```diff
@@ -0,0 +1,245 @@
+	return client
+}
+
+func getPrimaryKeyType(primaryKeyType string) tablestore.PrimaryKeyType{
+	var keyType tablestore.PrimaryKeyType
+	switch primaryKeyType{
+	case "INTEGER":
+		keyType = tablestore.PrimaryKeyType_INTEGER
+	case "STRING":
+		keyType = tablestore.PrimaryKeyType_STRING
+	case "BINARY":
+		keyType = tablestore.PrimaryKeyType_BINARY
+	}
+	return keyType
+}
+
+func resourceAliyunOtsTableCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("Creating OTS Table...")
+	client := getOtsClient()
+
+	tableMeta := new(tablestore.TableMeta)
+
+	table_name := d.Get("table_name").(string)
+	tableMeta.TableName = table_name
+
+	primaryKeyType := d.Get("primary_key_1_type").(string)
+
+	keyType := getPrimaryKeyType(primaryKeyType)
+
+	tableMeta.AddPrimaryKeyColumn(d.Get("primary_key_1_name").(string), keyType)
+
+	primaryKey2Name := d.Get("primary_key_2_name")
+	primaryKey2Type := getPrimaryKeyType(d.Get("primary_key_2_type").(string))
+	//Whether need to check the value of primaryKey2Type?
+	if primaryKey2Name != nil {
+		tableMeta.AddPrimaryKeyColumn(primaryKey2Name.(string), primaryKey2Type)
+	}
+
+	primaryKey3Name := d.Get("primary_key_3_name")
+	primaryKey3Type := getPrimaryKeyType(d.Get("primary_key_3_type").(string))
+	if primaryKey3Name != nil {
+		tableMeta.AddPrimaryKeyColumn(primaryKey3Name.(string), primaryKey3Type)
+	}
+
+	primaryKey4Name := d.Get("primary_key_4_name")
+	primaryKey4Type := getPrimaryKeyType(d.Get("primary_key_4_type").(string))
+	if primaryKey4Name != nil {
+		tableMeta.AddPrimaryKeyColumn(primaryKey4Name.(string), primaryKey4Type)
+	}
+
+	tableOption := new(tablestore.TableOption)
+	tableOption.TimeToAlive = d.Get("time_to_live").(int)
+	tableOption.MaxVersion = d.Get("max_version").(int)
+	reservedThroughput := new(tablestore.ReservedThroughput)
+	//reservedThroughput.Readcap = d.Get("read_cap").(int)
+	//reservedThroughput.Writecap = d.Get("write_cap").(int)
+
+	createTableRequest := new(tablestore.CreateTableRequest)
+	createTableRequest.TableMeta = tableMeta
+	createTableRequest.TableOption = tableOption
+	createTableRequest.ReservedThroughput = reservedThroughput
+	_, err := client.CreateTable(createTableRequest)
+	if err != nil {
+		return fmt.Errorf("Failed to create table with error: %s", err)
+	} else {
+		log.Printf("Create table finished")
+	}
+
+	// Need to set id before calling read method or terraform.state won't be generated.
+	d.SetId(table_name)
+	//return resourceAliyunOtsTableUpdate(d, meta)
+	return resourceAliyunOtsTableRead(d, meta)
+}
+
+func resourceAliyunOtsTableRead(d *schema.ResourceData, meta interface{}) error {
+	log.Println("Describing OTS Table...")
+	client := getOtsClient()
+
+	describeTableReq := new(tablestore.DescribeTableRequest)
+	describeTableReq.TableName = d.Get("table_name").(string)
+
+	describ, err := client.DescribeTable(describeTableReq)
+
+	if err != nil {
+		fmt.Println("failed to update table with error: %s", err)
```

Comment:
If there is a error, it must return directly.

Reply:
Done with all similar ones.